[svg] fix smart pointer warnings introduced in 280621@main

graouts opened the pull request:
#### db4bb37b38dcfbedf24bac1d352094c240b53131
<pre>
[svg] fix smart pointer warnings introduced in 280621@main
<a href="https://bugs.webkit.org/show_bug.cgi?id=276217">https://bugs.webkit.org/show_bug.cgi?id=276217</a>
<a href="https://rdar.apple.com/131103106">rdar://131103106</a>

Reviewed by Charlie Wolfe.

* Source/WebCore/svg/SVGPathElement.cpp:
(WebCore::SVGPathElement::path const):
(WebCore::SVGPathElement::collectPresentationalHintsForAttribute):

Canonical link: <a href="https://commits.webkit.org/280751@main">https://commits.webkit.org/280751@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1069ba6149cb4c3e095e1a10712758fc944c591c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/38/builds/57500 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/36828 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/9975 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/61122 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/59/builds/7945 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/9/builds/59628 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/44461 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/8133 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/61122 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/59/builds/7945 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/59530 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/34558 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/49672 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/61122 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/31341 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/6975 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/6948 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/53307 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/7247 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/62801 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/87/builds/1413 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/7335 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/62801 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/86/builds/1420 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/49697 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/62801 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8581 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/32657 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/33742 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/34827 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/33488 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

charliewolfe commented on Source/WebCore/svg/SVGPathElement.cpp:
```diff
@@ -223,22 +223,22 @@ RenderPtr<RenderElement> SVGPathElement::createElementRenderer(RenderStyle&& sty
         // In the case of the `d` property, we want to avoid providing a string value since it will require
         // the path data to be parsed again and path data can be unwieldy.
-        auto property = cssPropertyIdForSVGAttributeName(name, document().settings());
+        auto property = cssPropertyIdForSVGAttributeName(name, protectedDocument()->settings());
```
I think the static analyzer wants the settings object to be protected here, not document.

graouts replied:
Do I need to create a local `Ref` or can I just call the method like this?

```c++
auto property = cssPropertyIdForSVGAttributeName(name, Ref { document().settings() });
```

charliewolfe commented on Source/WebCore/svg/SVGPathElement.cpp:
```diff
@@ -223,22 +223,22 @@ RenderPtr<RenderElement> SVGPathElement::createElementRenderer(RenderStyle&& sty
         // In the case of the `d` property, we want to avoid providing a string value since it will require
         // the path data to be parsed again and path data can be unwieldy.
-        auto property = cssPropertyIdForSVGAttributeName(name, document().settings());
+        auto property = cssPropertyIdForSVGAttributeName(name, Ref { document().settings() });
```
The style I usually see is to call `protectedSettings()` on Document here, but `Ref { x }` shows up often too, so maybe that’s also fine. @rniwa or @cdumez might have an opinion.

graouts replied:
Did not know about `protectedSettings()`! I'll adopt whatever the braintrust suggests.

rniwa replied:
Yeah, the preferred style here is to call `protectedSettings()` here. An explicit instantiation of `Ref` or `RefPtr` is useful when we're allocating a local variable. In the cases where we use the result of a function call to call another function or chain a function call (e.g. `document().settings().nonTrivialFunction()`), a preferred method is to use `protectedX` function.

graouts committed:
[svg] fix smart pointer warnings introduced in 280621@main
https://bugs.webkit.org/show_bug.cgi?id=276217
rdar://131103106

Reviewed by Charlie Wolfe.

* Source/WebCore/svg/SVGPathElement.cpp:
(WebCore::SVGPathElement::path const):
(WebCore::SVGPathElement::collectPresentationalHintsForAttribute):

Canonical link: https://commits.webkit.org/280751@main
